matplotcheck/base.py

```diff
@@ -675,50 +678,58 @@ def assert_xydata(
         None :
             Nothing if no legends overlap, otherwise throws ``AssertionError`` with message `m`
         """
+        if xy_expected is None:
+            return
+        elif not isinstance(xy_expected, pd.DataFrame):
+            raise ValueError(
+                "xy_expected must be of type: pandas dataframe or Geopandas Dataframe"
+            )
 
-        # If there data are spatial (geopandas), grab geometry data
-        if type(xy_expected) == gpd.geodataframe.GeoDataFrame and not xcol:
+        # If xy_expected is a GeoDataFrame, then we make is a normal DataFrame
+        # with the coordinates of the geometry in that GeoDataFrame as the x and
+        # y data
+        if isinstance(xy_expected, gpd.geodataframe.GeoDataFrame) and not xcol:
             xy_expected = pd.DataFrame(
                 data={
                     "x": [p.x for p in xy_expected.geometry],
                     "y": [p.y for p in xy_expected.geometry],
                 }
             ).dropna()
             xcol, ycol = "x", "y"
-        if (
-            type(xy_expected) == pd.DataFrame
-            or type(xy_expected) == gpd.geodataframe.GeoDataFrame
-        ):
-            if xlabels:
-                self.assert_xlabel_ydata(xy_expected, xcol=xcol, ycol=ycol)
-                return
-            xy_data = self.get_xy(points_only=points_only, xtime=xtime)
-
-            # Make sure the data are sorted the same
-            xy_data, xy_expected = (
-                xy_data.sort_values(by="x"),
-                xy_expected.sort_values(by=xcol),
-            )
-            if tolerence > 0:
-                if xtime:
-                    raise ValueError(
-                        "tolerance must be 0 with datetime on x-axis"
-                    )
+
+        if xlabels:
+            self.assert_xlabel_ydata(xy_expected, xcol=xcol, ycol=ycol, m=m)
+            return
+        xy_data = self.get_xy(points_only=points_only, xtime=xtime)
+
+        # Make sure the data are sorted the same
+        xy_data, xy_expected = (
+            xy_data.sort_values(by="x"),
+            xy_expected.sort_values(by=xcol),
+        )
+
+        if tolerence > 0:
+            if xtime:
+                raise ValueError("tolerance must be 0 with datetime on x-axis")
+            """
+            for i in range(100):
                 np.testing.assert_allclose(
-                    xy_data["x"], xy_expected[xcol], rtol=tolerence, err_msg=m
+                    xy_data.iloc[0, 0:i], xy_expected.iloc[0, 0:i], rtol=tolerence, err_msg=m
```

Comment:
hey @ryla5068 can you explain why some of these changes were implemented? what is hard for me to see if why the change was made given many of the tests are new! i'm guessing it's because pandas requires iloc now to call cols and rows but here you are using sequential calls rather than the entire column